app/src/main/java/com/kickstarter/viewmodels/CommentsViewModel.kt

```diff
@@ -123,42 +139,81 @@ interface CommentsViewModel {
 
             loadCommentList(initialProject)
 
-            this.currentUser.loggedInUser()
+            commentsList
+                .compose<Pair<List<CommentCardData>, User >>(combineLatestPair(this.currentUser.loggedInUser()))
                 .compose(Transformers.takePairWhen(this.insertNewCommentToList))
                 .map {
-                    buildCommentBody(it)
+                    Pair(it.first.first, buildCommentBody(Pair(it.first.second, it.second)))
                 }
-                .compose<Pair<Comment, Project?>>(combineLatestPair(initialProject))
+                .compose(combineLatestPair(initialProject))
                 .map {
-                    CommentCardData.builder()
-                        .comment(it.first)
-                        .project(it.second)
-                        .commentCardState(CommentCardStatus.TRYING_TO_POST.commentCardStatus)
-                        .build()
+                    Pair(
+                        it.first.first,
+                        CommentCardData.builder()
+                            .comment(it.first.second)
+                            .project(it.second)
+                            .commentCardState(CommentCardStatus.TRYING_TO_POST.commentCardStatus)
+                            .build()
+                    )
+                }
+                    .doOnNext { scrollToTop.onNext(true) }
+                .compose(bindToLifecycle())
+                .subscribe {
+                    it.first.toMutableList().apply {
+                        add(0, it.second)
+                        commentsList.onNext(this)
+                    }
+                }
+
+            this.onShowGuideLinesLinkClicked
+                .compose(bindToLifecycle())
+                .subscribe {
+                    showGuideLinesLink.onNext(null)
                 }
+
+            this.commentsList
+                .map { it.size }
+                .distinctUntilChanged()
                 .compose(bindToLifecycle())
                 .subscribe {
-                    this.insertComment.onNext(it)
+                    this.setEmptyState.onNext(it == 0)
+                }
+
+            // TODO showcasing subscription to initialization to be completed on : https://kickstarter.atlassian.net/browse/NT-1951
```

Comment:
@sunday-okpoluaefe  the error logic is ready for https://kickstarter.atlassian.net/browse/NT-1951